2015-11-16  Charles Baylis  <charles.baylis@linaro.org>

	Backport from mainline r227407
	PR ipa/67280
	* cgraphunit.c (cgraph_node::create_wrapper): Set can_throw_external
	in new callgraph edge.

--- a/src/gcc/cgraphunit.c
+++ b/src/gcc/cgraphunit.c
@@ -2505,7 +2505,6 @@
   memset (&thunk, 0, sizeof (cgraph_thunk_info));
   thunk.thunk_p = true;
   create_edge (target, NULL, count, CGRAPH_FREQ_BASE);
-  callees->can_throw_external = !TREE_NOTHROW (target->decl);
 
   tree arguments = DECL_ARGUMENTS (decl);
 
